api/admin/put_settings.js

```diff
@@ -3,28 +3,30 @@
 const Joi = require('@hapi/joi');
 const storage = require('../../lib/storage');
 
-const urlRegex = /https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)/;
+const logoPathRegex = /https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)/;
 const colorRegex = /^#[A-Fa-f0-9]{6}/;
 
+// For validation, keep existing required
+// fields, but let users send a customDomain
+// only if they'd like
 module.exports = () => ({
   method: 'PUT',
   options: {
     auth: {
       strategies: ['jwt']
     },
     validate: {
-      payload: {
-        template: Joi.string().required(),
-        locale: Joi.string().required(),
-        title: Joi.string().required(),
-        color: Joi.string()
-          .regex(colorRegex)
-          .required(),
+      payload: Joi.object({
+        template: Joi.string(),
+        locale: Joi.string(),
+        title: Joi.string(),
+        color: Joi.string().regex(colorRegex),
         logoPath: Joi.string()
-          .regex(urlRegex)
+          .regex(logoPathRegex)
           .allow(''),
-        removeOverlay: Joi.bool().default(false)
-      }
+        removeOverlay: Joi.bool().default(false),
+        customDomain: Joi.string().allow('')
```

Comment:
wanna add a code comment that empty strings are used to unset the custom domain?

Reply:
I moved it next to allow('')